usaspending_api/recipient/v2/views/recipients.py

```diff
@@ -225,60 +231,85 @@ def extract_business_categories(recipient_name, recipient_duns, recipient_hash):
     return sorted(business_categories)
 
 
-def obtain_recipient_totals(recipient_id, children=False, year="latest", subawards=False):
+def obtain_recipient_totals(recipient_id, children=False, year="latest"):
     """ Extract the total amount and transaction count for the recipient_hash given the timeframe
 
         Args:
             recipient_id: string of hash(duns, name)-[recipient-level]
             children: whether or not to group by children
             year: the year the totals/counts are based on
-            subawards: whether to total based on subawards
         Returns:
             list of dictionaries representing hashes and their totals/counts
     """
-    if year == "latest" and children is False:
-        # Simply pull the total and count from RecipientProfile
-        recipient_hash = recipient_id[:-2]
-        recipient_level = recipient_id[-1]
-        results = list(
-            RecipientProfile.objects.filter(recipient_hash=recipient_hash, recipient_level=recipient_level)
-            .annotate(total=F("last_12_months"), count=F("last_12_months_count"))
-            .values("recipient_hash", "recipient_unique_id", "recipient_name", "total", "count")
-        )
+    filters = reshape_filters(recipient_id=recipient_id, year=year)
+    filter_query = QueryWithFilters.generate_transactions_elasticsearch_query(filters)
+
+    search = TransactionSearch().filter(filter_query)
 
+    if children:
+        group_by_field = "recipient_agg_key"
+    elif recipient_id[-2:] == "-P":
+        group_by_field = "parent_recipient_hash"
     else:
-        filters = reshape_filters(recipient_id=recipient_id, year=year)
-        queryset, model = recipient_totals(filters)
+        group_by_field = "recipient_hash"
+
+    bucket_count = get_number_of_unique_terms(filter_query, f"{group_by_field}.hash")
+
+    if bucket_count == 0:
+        return []
+
+    group_by_recipient = A("terms", field=group_by_field, size=bucket_count, shard_size=bucket_count)
+    sum_obligation = get_scaled_sum_aggregations("generated_pragmatic_obligation")["sum_field"]
+    sum_face_value_loan = get_scaled_sum_aggregations("face_value_loan_guarantee")["sum_field"]
+    obligation_count = A("value_count", field="generated_pragmatic_obligation")
```

Comment:
I'm unsure if counting the values will be accurate. I see above that nulls are replacing 0 in the index. But 0 is a valid amount for transactions (definitely true for non-loans' `federal_action_obligation`, need to check loans' data to 100% confirm)

Is there another way to get the counts?

Reply:
Not 100% sure, would need to check.

Reply:
Count of `transaction_id` for total transactions?
Count of type in `07`, `08` for total loans?

Reply:
Change the ES query to filter by type "07" and "08" before doing the sum to get an accurate count. Due to the small number of records, this is still sub one second when I run this aggregation against Non-Prod cluster.